src/compiler/symb-mangling.bas

```diff
@@ -451,6 +452,17 @@ sub symbMangleType _
 		dtype = typeJoin( dtype and (not FB_DATATYPE_INVALID), FB_DATATYPE_STRUCT )
 	end if
 
+	'' const array?
+	if( typeIsConst( dtype ) ) then
+		if( (options and FB_MANGLEOPT_KEEPTOPCONST) <> 0 ) then
```

Comment:
Maybe the toplevel const checks in this function should both be moved between the ref and ptr checks, such that:
- "reference to const" will only mangle the const once (avoiding `KRK`) even with FB_MANGLEOPT_KEEPTOPCONST
- allow mangling "const foo const ptr" as `KPK` (probably not valid for C++?) by handling the toplevel const (dropping or mangling) before recursing in the ptr case

Reply:
I see, OK.  I think you're correct.

Looks like `KPK` is valid for C++, as in this example will mangle the parameter to KPKi
```
template <class T>
T P (T a) {
 return a;
}

void proc()
{
	int const x = 5;
	int const* const y = &x;
	int const* const z = P <int const* const> (y);
}
```

`c++filt` will demangle `KRKi` to `int const& const` though looks like there's no legal way to ever use that in C++.